### PR TITLE
Pfappserver Updates

### PR DESCRIPTION
**Pull request by satkunas**
# Description
Continue work on the pfappserver

# Impacts
Increased PR count :100: 

# Delete branch after merge
YES

### CODE REVIEW

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 1]
```diff
@@ -6,9 +6,55 @@
         <br/>
         <b-container fluid class="mt-3 px-0 text-right">
           <b-button type="reset" variant="outline-secondary">{{ $t('Reset') }}</b-button>
-          <b-button type="submit" variant="outline-primary">{{ $t('Search') }}</b-button>
+          <b-button-group>
+            <b-button type="submit" variant="outline-primary">{{ $t('Search') }}</b-button>
+            <b-dropdown variant="outline-primary" right>
+              <b-dropdown-item @click="showSaveSearchModal=true">
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue]: Why not using the [v-b-modal](https://bootstrap-vue.js.org/docs/components/modal/#using-v-b-modal-directive-recommended-) directive?

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 2]
```diff
@@ -6,9 +6,55 @@
...
+                <icon class="position-absolute mt-1" name="save"></icon>
+                <span class="ml-4">{{ $t('Save Search') }}</span>
+              </b-dropdown-item>
+              <b-dropdown-divider></b-dropdown-divider>
+              <b-dropdown-item @click="showExportJsonModal=true">
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue:17]: Why not using the [v-b-modal](https://bootstrap-vue.js.org/docs/components/modal/#using-v-b-modal-directive-recommended-) directive?

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 3]
```diff
@@ -6,9 +6,55 @@
...
+                <icon class="position-absolute mt-1" name="sign-out-alt"></icon>
+                <span class="ml-4">{{ $t('Export to JSON') }}</span>
+              </b-dropdown-item>
+              <b-dropdown-item @click="showImportJsonModal=true">
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue:21]: Why not using the [v-b-modal](https://bootstrap-vue.js.org/docs/components/modal/#using-v-b-modal-directive-recommended-) directive?

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue [hunk 4]
```diff
@@ -74,8 +144,41 @@ export default {
       this.$emit('submit-search', query)
     },
     onReset (event) {
-      this.condition = { op: 'and', values: [{ op: 'or', values: [{ field: this.fields[0].value, op: null, value: null }] }] }
+      this.condition = this.defaultCondition()
       this.$emit('reset-search')
+    },
+    copyNodeExportJsonTextarea () {
+      this.$refs.nodeExportJsonTextarea.$el.select()
+      document.execCommand('copy')
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/components/pfSearch.vue]: Make sure the command is available using `document.queryCommandSupported('copy')`.

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue [hunk 5]
```diff
@@ -29,6 +33,22 @@
 
 <script>
 export default {
-  name: 'Nodes'
+  name: 'Nodes',
+  data () {
+    return {}
+  },
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: You can remove the `data` block.

File: html/pfappserver/root/static.alt/src/views/Nodes/index.vue [hunk 6]
```diff
@@ -29,6 +33,22 @@
...
+    savedSearches () {
+      return this.$store.state.$_nodes.savedSearches
+    }
+  },
+  methods: {
+    deleteSavedSearch (search) {
+      this.$store.dispatch('$_nodes/deleteSavedSearch', search)
+    },
+    urlSavedSearch (search) {
+      return '/nodes/search?query=' + JSON.stringify(search.query)
```
Comment (cgx) [inline html/pfappserver/root/static.alt/src/views/Nodes/index.vue]: Use the `this.$router` instance instead.

**Review by cgx — CHANGES_REQUESTED** [review]

File: html/pfappserver/root/static.alt/src/components/pfSearch.vue (hunk 2 above)
Comment (satkunas) [reply html/pfappserver/root/static.alt/src/components/pfSearch.vue:17]: There needed to be more granular control on when/how to close the modals (alt methods), thus the simpler v-b-modal wasn't sufficient.

**Review by cgx — APPROVED** [review]

**Pull request merged**